pkg/R/DataFrame.R

```diff
@@ -0,0 +1,119 @@
+# DataFrame.R - DataFrame class and methods implemented in S4 OO classes
+
+#' @include jobj.R RDD.R
+NULL
+
+setOldClass("jobj")
+
+#' @title S4 class that represents a DataFrame
+#' @description DataFrames can be created using functions like
+#'              \code{jsonFile}, \code{table} etc.
+#' @rdname DataFrame
+#' @seealso jsonFile, table
+#'
+#' @param env An R environment that stores bookkeeping states of the DataFrame
+#' @param sdf A Java object reference to the backing Scala SchemaRDD
+#' @export
+
+setClass("DataFrame",
+         slots = list(env = "environment",
+                      sdf = "jobj"))
+
+setMethod("initialize", "DataFrame", function(.Object, sdf, isCached, isCheckpointed) {
+  .Object@env <- new.env()
+  .Object@env$isCached <- isCached
+  .Object@env$isCheckpointed <- isCheckpointed
+  
+  .Object@sdf <- sdf
+  .Object
+})
+
+#' @rdname DataFrame
+#' @export
+
+dataFrame <- function(sdf, isCached = FALSE, isCheckpointed = FALSE) {
+  new("DataFrame", sdf, isCached, isCheckpointed)
+}
+
+# The DataFrame accessor function
+
+setGeneric("getsdf", function(df, ...) {standardGeneric("getsdf") })
+setMethod("getsdf", signature(df = "DataFrame" ), function(df) df@sdf )
+
+############################ DataFrame Methods ##############################################
+
+#' Print Schema of a DataFrame
+#' 
+#' Prints out the schema in tree format
+#' 
+#' @param df A SparkSQL DataFrame
+#' 
```

Comment:
A general comment for all exported functions is that it would be good to include examples here. You can wrap the example in `dontrun` if there are external dependencies like having `sc` around etc. The examples are very useful for figuring out what the operator looks like etc.